go/saltpack/encrypt.go

```diff
@@ -149,17 +150,19 @@ func (es *encryptStream) init(sender BoxSecretKey, receivers []BoxPublicKey) err
 
 	eh.SenderSecretbox = secretbox.Seal([]byte{}, sender.GetPublicKey().ToKID(), (*[24]byte)(nonceForSenderKeySecretBox()), (*[32]byte)(&es.payloadKey))
 
-	for _, receiver := range receivers {
-		payloadKeyBox := ephemeralKey.Box(receiver, nonceForPayloadKeyBox(), es.payloadKey[:])
+	if !es.anonymous {
+		for _, receiver := range receivers {
+			payloadKeyBox := ephemeralKey.Box(receiver, nonceForPayloadKeyBox(), es.payloadKey[:])
 
-		keys := receiverKeys{PayloadKeyBox: payloadKeyBox}
+			keys := receiverKeys{PayloadKeyBox: payloadKeyBox}
 
-		// Don't specify the receivers if this public key wants to hide
-		if !receiver.HideIdentity() {
-			keys.ReceiverKID = receiver.ToKID()
-		}
+			// Don't specify the receivers if this public key wants to hide
+			if !receiver.HideIdentity() {
+				keys.ReceiverKID = receiver.ToKID()
```

Comment:
Exactly. Which wouldn't be too bad since the `naclBoxPublicKey` type is a throw-away shim around the key. So you can introduce a "supertype" in `libkb/saltpack.go` that is just of the form `hiddenNaclBoxPublicKey` and create those rather than `naclBoxPublicKey` in `SaltpackEncrypt` line 26.